pkg/ottl/context_inferrer.go

```diff
@@ -81,11 +83,38 @@ func withContextInferrerPriorities(priorities []string) priorityContextInferrerO
 	}
 }
 
-func (s *priorityContextInferrer) infer(statements []string) (inferredContext string, err error) {
-	s.telemetrySettings.Logger.Debug("Inferring context from statements",
+func (s *priorityContextInferrer) inferFromConditions(conditions []string) (inferredContext string, err error) {
+	return s.innerInfer(conditions, func(condition string) ([]path, map[string]struct{}, map[enumSymbol]struct{}, error) {
```

Comment:
I think we could clean this up a bit more by moving the parsing logic into the `getParsedConditionHints` (renaming it to `getConditionHints`), so it would match the function signature, and could be used here as `return s.innerInfer(statements, s.getStatementHints)`.